subsys/bluetooth/services/bas/bas_bls.c

```diff
@@ -194,6 +194,23 @@ void bt_bas_bls_set_battery_charge_level(enum bt_bas_bls_battery_charge_level le
 	bls.power_state &= ~BATTERY_CHARGE_LEVEL_MASK;
 	bls.power_state |= (level << BATTERY_CHARGE_LEVEL_SHIFT) & BATTERY_CHARGE_LEVEL_MASK;
 	bt_bas_bls_update_battery_level_status();
+
+	/* Return if CONFIG_BT_BAS_BCS is not enabled. */
+	if (!IS_ENABLED(CONFIG_BT_BAS_BCS)) {
+		return;
+	}
+
+	/*
+	 * Set the BCS Critical Power State bit as per BAS 1.1 specification section 3.4.1.1:
+	 * The BCS Critical Power State bit should be set to true if the Battery Charge Level
+	 * is set to Critical in the Power State field.
+	 */
+	if (level == BT_BAS_BLS_CHARGE_LEVEL_CRITICAL) {
+		bt_bas_bcs_set_battery_critical_state(true);
+		return;
+	}
+
+	bt_bas_bcs_set_battery_critical_state(false);
```

Comment:
If `value == BT_BAS_BLS_CHARGE_LEVEL_UNKNOWN` should we really modify the state?

Reply:
You are correct, if battery is set to critical, then there shouldn't be an option to reset.
But in the case of testing , this reset will be handy.

Reply:
> You are correct, if battery is set to critical, then there shouldn't be an option to reset.

That is not what I meant :) 

And also incorrect; it should definitely be possible to reset the battery state from critical to non-critical, e.g. if changing the battery.